### PR TITLE
[Spark] Disable security context enrichment for remote spark [1.0.x]

### PR DESCRIPTION
**Pull request by yaelgen**
fix for: https://jira.iguazeng.com/browse/ML-2467

- Added validation on server side when enriching security context
- Raising exception when trying to set security context for remote spark

### CODE REVIEW

File: mlrun/api/api/utils.py [hunk 1]
```diff
@@ -486,6 +486,7 @@ def ensure_function_security_context(function, auth_info: mlrun.api.schemas.Auth
         == SecurityContextEnrichmentModes.disabled.value
         or mlrun.runtimes.RuntimeKinds.is_local_runtime(function.kind)
         or function.kind == mlrun.runtimes.RuntimeKinds.spark
+        or function.kind == mlrun.runtimes.RuntimeKinds.remotespark
```
Comment (Tankilevitch) [inline mlrun/api/api/utils.py:489]: Please add a comment why remote spark is here

**Review by Tankilevitch — CHANGES_REQUESTED** [review]
Looks good, minor comment

File: mlrun/api/api/utils.py [hunk 2]
```diff
@@ -486,6 +486,10 @@ def ensure_function_security_context(function, auth_info: mlrun.api.schemas.Auth
         == SecurityContextEnrichmentModes.disabled.value
         or mlrun.runtimes.RuntimeKinds.is_local_runtime(function.kind)
         or function.kind == mlrun.runtimes.RuntimeKinds.spark
+        # We do not enrich remote spark security context,
+        # as its image contains dirs and files with user 1000(iguazio) as owner,
+        # so the running user must either be 1000 or root
```
Comment (Tankilevitch) [inline mlrun/api/api/utils.py:491]: Mainly a style manner, wdyt?
```suggestion
        # We do not enrich remote spark security context,
        # as its image currently supports running only with user 1000 or root
        # and by default it runs with user 1000 ( even when not setting security context )
```

File: mlrun/api/api/utils.py [hunk 3]
```diff
@@ -486,6 +486,10 @@ def ensure_function_security_context(function, auth_info: mlrun.api.schemas.Auth
         == SecurityContextEnrichmentModes.disabled.value
         or mlrun.runtimes.RuntimeKinds.is_local_runtime(function.kind)
         or function.kind == mlrun.runtimes.RuntimeKinds.spark
+        # We do not enrich remote spark security context,
+        # as its image currently supports running only with user 1000 or root
+        # and by default it runs with user 1000 (even when not setting security context)
```
Comment (AlonMaor14) [inline mlrun/api/api/utils.py:491]:
```suggestion
        # remote spark image currently requires running with user 1000 or root
        # and by default it runs with user 1000 (when security context is not set)
```

File: mlrun/runtimes/remotesparkjob.py [hunk 4]
```diff
@@ -144,6 +147,18 @@ def with_spark_service(self, spark_service, provider=RemoteSparkProviders.iguazi
...
+    def with_security_context(
+        self, security_context: kubernetes.client.V1SecurityContext
+    ):
+        """
+        With security context is not supported for spark runtime.
+        Driver / Executor processes run with uid / gid 1000 as long as security context is not defined.
+        If in the future we want to support setting security context it will work only from spark version 3.2 onwards.
+        """
+        raise mlrun.errors.MLRunInvalidArgumentTypeError(
+            "with_security_context is not supported with spark operator"
```
Comment (AlonMaor14) [inline mlrun/runtimes/remotesparkjob.py:159]:
```suggestion
            "with_security_context is not supported with remote spark"
```

**Review by AlonMaor14 — COMMENTED** [review]
Looks good 👍 
Minor suggestions

**Review by Tankilevitch — APPROVED** [review]

**Pull request merged**